### PR TITLE
rest API: access restriction by organisation read, write, delete, update

### PR DESCRIPTION
**Pull request by BadrAly**
* NEW Restricts access to APIs for users of same organisation.

Signed-off-by: Aly Badr <aly.badr@rero.ch>

* ./run_tests.sh
* localhost:5000/api/< record_type >/ < pid > 
read/write/delete/update is now available only for records of logged users organisation

### CODE REVIEW

File: tests/api/test_circ_policies_rest.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def test_circ_policies_permissions(
         data={},
         headers=json_header
     )
-    assert res.status_code == 401
```
Comment (jma) [inline tests/api/test_circ_policies_rest.py]: to check

File: tests/api/test_circ_policies_rest.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def test_circ_policies_permissions(
     )
 
     res = client.delete(item_url)
-    assert res.status_code == 403
+    assert res.status_code == 401
```
Comment (jma) [inline tests/api/test_circ_policies_rest.py]: to check

File: tests/api/test_items_rest_views.py [hunk 3]
```diff
@@ -413,3 +414,102 @@ class locale:
...
+
+    # Sion
+    login_user_via_session(client, librarian_sion_no_email.user)
+
+    res = client.post(
+        record_url,
+        data=json.dumps(item_lib_saxon),
+        headers=json_header
+    )
+    assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_items_rest_views.py]: probably bad url

File: tests/api/test_libraries_rest.py [hunk 4]
```diff
@@ -243,3 +243,104 @@ def test_filtered_libraries_get(
...
+
+    # Sion
+    login_user_via_session(client, librarian_sion_no_email.user)
+
+    res = client.post(
+        record_url,
+        data=json.dumps(lib_fully_data),
+        headers=json_header
+    )
+    assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_libraries_rest.py]: bad url

File: tests/api/test_locations_rest.py [hunk 5]
```diff
@@ -203,3 +206,103 @@ def test_filtered_locations_get(
...
+
+    # Sion
+    login_user_via_session(client, librarian_sion_no_email.user)
+
+    res = client.post(
+        record_url,
+        data=json.dumps(loc_restricted_saxon_data),
+        headers=json_header
+    )
+    assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_locations_rest.py]: bad url

File: tests/api/test_organisations_rest_api.py [hunk 6]
```diff
@@ -23,3 +30,97 @@ def test_location_can_delete(client, org_martigny, lib_martigny):
...
+
+    # Sion
+    login_user_via_session(client, librarian_sion_no_email.user)
+
+    res = client.post(
+        record_url,
+        data=json.dumps(org_martigny_data),
+        headers=json_header
+    )
+    assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_organisations_rest_api.py]: bad url

File: tests/api/test_patron_types_rest.py [hunk 7]
```diff
@@ -222,3 +225,105 @@ def test_filtered_patron_types_get(
...
+
+    # Sion
+    login_user_via_session(client, librarian_sion_no_email.user)
+
+    res = client.post(
+        record_url,
+        data=json.dumps(patron_type_adults_martigny_data),
+        headers=json_header
+    )
+    assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_patron_types_rest.py]: bad url

File: tests/api/test_patrons_rest.py [hunk 8]
```diff
@@ -187,3 +189,103 @@ def test_patrons_post_put_delete(client, lib_martigny,
...
+
+    # Sion
+    # login_user_via_session(client, librarian_sion_no_email.user)
+
+    # res = client.post(
+    #     record_url,
+    #     data=json.dumps(patron_martigny_data),
+    #     headers=json_header
+    # )
+    # assert res.status_code == 405
```
Comment (jma) [inline tests/api/test_patrons_rest.py]: change even if it is in the comment

File: tests/api/test_items_rest_views.py (hunk 3 above)
Comment (BadrAly) [reply tests/api/test_items_rest_views.py]: fixed

File: tests/api/test_circ_policies_rest.py (hunk 2 above)
Comment (BadrAly) [reply tests/api/test_circ_policies_rest.py]: 401 Unauthorized cause we do not have a logger user here.

File: tests/api/test_circ_policies_rest.py (hunk 1 above)
Comment (BadrAly) [reply tests/api/test_circ_policies_rest.py]: Most likely because `data={}`?

Comment (jma) [reply tests/api/test_circ_policies_rest.py]: yes probably

File: tests/api/test_item_types_rest.py [hunk 9]
```diff
@@ -45,7 +48,7 @@ def test_item_types_permissions(client, item_type_standard_martigny,
         data={},
         headers=json_header
     )
-    assert res.status_code == 401
+    assert res.status_code == 400
```
Comment (jma) [inline tests/api/test_item_types_rest.py]: bad status here, as nobody is logged in we should have a 401. Needs some check. Once it is solved please check similar cases.

File: tests/api/test_circ_policies_rest.py (hunk 1 above)
Comment (BadrAly) [reply tests/api/test_circ_policies_rest.py]: fixed

File: tests/api/test_item_types_rest.py (hunk 9 above)
Comment (BadrAly) [reply tests/api/test_item_types_rest.py]: fixed

**Comment by BadrAly** [comment]
> Please remove the parenthesis from the commit message title.
> Else, this PR is OK for me, I even tested the professional views and didn't identified issues.

done

**Pull request merged**